src/EnergyPlus/PoweredInductionUnits.cc

```diff
@@ -266,473 +266,304 @@ void GetPIUs(EnergyPlusData &state)
-
-        state.dataInputProcessing->inputProcessor->getObjectItem(state,
-                                                                 cCurrentModuleObject,
-                                                                 PIUIndex,
-                                                                 state.dataIPShortCut->cAlphaArgs,
-                                                                 NumAlphas,
-                                                                 state.dataIPShortCut->rNumericArgs,
-                                                                 NumNumbers,
-                                                                 IOStatus,
-                                                                 state.dataIPShortCut->lNumericFieldBlanks,
-                                                                 state.dataIPShortCut->lAlphaFieldBlanks,
-                                                                 state.dataIPShortCut->cAlphaFieldNames,
-                                                                 state.dataIPShortCut->cNumericFieldNames);
-
-        GlobalNames::VerifyUniqueInterObjectName(state,
-                                                 state.dataPowerInductionUnits->PiuUniqueNames,
-                                                 state.dataIPShortCut->cAlphaArgs(1),
-                                                 cCurrentModuleObject,
-                                                 state.dataIPShortCut->cAlphaFieldNames(1),
-                                                 ErrorsFound);
-        auto &thisPIU = state.dataPowerInductionUnits->PIU(PIUIndex);
-        thisPIU.Name = state.dataIPShortCut->cAlphaArgs(1);
-        thisPIU.UnitType = cCurrentModuleObject;
-        thisPIU.UnitType_Num = DataDefineEquip::ZnAirLoopEquipType::SingleDuct_SeriesPIU_Reheat;
-        thisPIU.Sched = state.dataIPShortCut->cAlphaArgs(2);
-        if (state.dataIPShortCut->lAlphaFieldBlanks(2)) {
-            thisPIU.SchedPtr = ScheduleManager::ScheduleAlwaysOn;
-        } else {
-            thisPIU.SchedPtr = GetScheduleIndex(state, state.dataIPShortCut->cAlphaArgs(2)); // convert schedule name to pointer
-            if (thisPIU.SchedPtr == 0) {
-                ShowSevereError(state,
-                                format("{}{}: invalid {} entered ={} for {}={}",
-                                       RoutineName,
-                                       cCurrentModuleObject,
-                                       state.dataIPShortCut->cAlphaFieldNames(2),
-                                       state.dataIPShortCut->cAlphaArgs(2),
-                                       state.dataIPShortCut->cAlphaFieldNames(1),
-                                       state.dataIPShortCut->cAlphaArgs(1)));
-                ErrorsFound = true;
-            }
-        }
+    for (const std::string cCurrentModuleObject : {"AirTerminal:SingleDuct:SeriesPIU:Reheat", "AirTerminal:SingleDuct:ParallelPIU:Reheat"}) {
```

Comment:
Refactoring of `GetPIUs`.

The existing approach uses the almost exact same code for series and parallel PIUs and relies field array indices to retrieve input values. Proposed approach uses a loop (decrease the code by almost half) and uses the field names instead of array indices to retrieve input values.